frontera/contrib/scrapy/schedulers/frontier.py

```diff
@@ -72,31 +72,43 @@ def _set_value(self, variable, value):
 
 
 class FronteraScheduler(Scheduler):
+    """
+    Custom Scheduler for Scrapy.
+    Adapts Frontera manager interface to Scrapy.
+
+    Important remarks:
+     - it doesn't enqueue majority of requests produced by middlewares or direct calls to engine, see self.enqueue_request
+       method, and override if needed,
+     - requires SchedulerSpiderMiddleware and SchedulerDownloaderMiddleware.
+    """
 
     def __init__(self, crawler, manager=None):
         self.crawler = crawler
         self.stats_manager = StatsManager(crawler.stats)
         self._pending_requests = deque()
-        self.redirect_enabled = crawler.settings.get('REDIRECT_ENABLED')
         settings = ScrapySettingsAdapter(crawler.settings)
         self.frontier = ScrapyFrontierManager(settings, manager)
         self._delay_on_empty = self.frontier.manager.settings.get('DELAY_ON_EMPTY')
+        self._redirect_enabled = crawler.settings.get('REDIRECT_ENABLED')
         self._delay_next_call = 0.0
-        self.logger = getLogger('frontera.contrib.scrapy.schedulers.FronteraScheduler')
+        self.logger = getLogger('frontera.contrib.scrapy.schedulers.frontier.FronteraScheduler')
 
     @classmethod
     def from_crawler(cls, crawler):
         return cls(crawler)
 
     def enqueue_request(self, request):
```

Comment:
Would it be even clearer to tag request that are enqueued in the Frontier through `process_spider_output -> links_extracted` (storing the tag in `meta` like you did for seed) and add any other non-tagged request to the local queue? This way you wouldn't need to check for redirect and other middleware could themselves chose between local scheduling or remote scheduling (Such as a middleware that logs in and try the request again without going through the frontier queue again).